### PR TITLE
ODIS scalability Improvements 

### PR DESCRIPTION
**Pull request by alecps**
Description
This PR adds a variety of improvements developed in response to the coinbase earn campaign. The most notable changes are 
- Improved async handling in the combiner
- added database timeouts 
- added timeouts to signer
- Fine-grained latency instrumentation added to signer, along with various other monitoring improvements 
- adds load test to monitor package

Other changes
Removes deprecated endpoint "getDistributedBlindedSalt" from combiner 

Tested
Most of these changes were deployed and tested e2e during coinbase earn. Any changes that have been added since are unlikely to cause issues and will be tested thoroughly before deployment as per the release process.

Related issues
NA

Backwards compatibility
Yes. Deprecated combiner endpoint "getDistributedBlindedSalt" removed but metrics show clients are no longer using it. 

Documentation
NA

### CODE REVIEW

File: packages/phone-number-privacy/signer/src/signing/get-partial-signature.ts [hunk 1]
```diff
@@ -196,3 +203,8 @@ function isValidGetSignatureInput(requestBody: GetBlindedMessagePartialSigReques
     hasValidTimestamp(requestBody)
   )
 }
+
+function isWhitelisted(requestBody: GetBlindedMessagePartialSigRequest) {
```
Comment (codyborn) [inline packages/phone-number-privacy/signer/src/signing/get-partial-signature.ts]: Nit
```suggestion
function considerWithQuotaForTesting(requestBody: GetBlindedMessagePartialSigRequest) {
```

File: .env.alfajores [hunk 2]
```diff
@@ -161,6 +161,7 @@ LOG_FORMAT=stackdriver
 # Options: fatal, error, warn, info (default), debug, trace
 LOG_LEVEL=info
 
+WHITELIST_PERCENTAGE=50
```
Comment (codyborn) [inline .env.alfajores]: Nit
```suggestion
TEST_QUOTA_BYPASS_PERCENTAGE=50
```

File: .env.alfajores [hunk 3]
```diff
@@ -293,4 +294,4 @@ AZURE_KOMENCI_WESTEU_KOMENCI_NETWORK=alfajores
 WALLET_CONNECT_IMAGE_REPOSITORY = 'us.gcr.io/celo-testnet/walletconnect',
 WALLET_CONNECT_IMAGE_TAG = '1472bcaad57e3746498f7a661c42ff5cf9acaf5a'
 WALLET_CONNECT_REDIS_CLUSTER_ENABLED = false
-WALLET_CONNECT_REDIS_CLUSTER_USEPASSWORD = false
```
Comment (alecps) [inline .env.alfajores:296]: does anyone know where this came from?

File: packages/phone-number-privacy/common/src/index.ts [hunk 4]
```diff
@@ -1,3 +1,4 @@
+export { retryAsyncWithBackOffAndTimeout, timeout } from '@celo/base'
```
Comment (alecps) [inline packages/phone-number-privacy/common/src/index.ts]: We probably don't need to do this anymore

File: packages/phone-number-privacy/signer/package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@celo/phone-number-privacy-signer",
-  "version": "1.1.4",
+  "version": "1.1.6",
```
Comment (alecps) [inline packages/phone-number-privacy/signer/package.json]: We should retag the latest image so this shows up in the status endpoint

File: packages/phone-number-privacy/signer/src/database/wrappers/account.ts [hunk 6]
```diff
@@ -81,16 +96,25 @@ export async function getDidMatchmaking(account: string, logger: Logger): Promis
...
  */
-export async function setDidMatchmaking(account: string, logger: Logger) {
+async function _setDidMatchmaking(account: string, logger: Logger) {
   logger.debug({ account }, 'Setting did matchmaking')
   try {
     if (await getAccountExists(account)) {
       return accounts()
         .where(ACCOUNTS_COLUMNS.address, account)
         .update(ACCOUNTS_COLUMNS.didMatchmaking, new Date())
+        .timeout(DB_TIMEOUT)
```
Comment (alecps) [inline packages/phone-number-privacy/signer/src/database/wrappers/account.ts]: This looks like a faulty merge. Shouldn't cause any problems I don't think but should be fixed

Comment (alecps) [reply packages/phone-number-privacy/signer/src/database/wrappers/account.ts]: @codyborn

File: packages/phone-number-privacy/signer/src/signing/query-quota.ts [hunk 7]
```diff
@@ -273,16 +292,21 @@ export async function getCeloBalance(logger: Logger, ...addresses: string[]): Pr
...
+  return retryAsyncWithBackOffAndTimeout(
+    async () => (await getContractKit().contracts.getAccounts()).getWalletAddress(account),
+    RETRY_COUNT,
+    [],
+    RETRY_DELAY_IN_MS,
+    undefined,
+    FULL_NODE_TIMEOUT_IN_MS
+  )
+    .catch((err: any) => {
+      logger.error({ account }, 'failed to get wallet address for account')
```
Comment (alecps) [inline packages/phone-number-privacy/signer/src/signing/query-quota.ts:314]: We should probably be recording a full node error here

Comment (alecps) [reply packages/phone-number-privacy/signer/src/signing/query-quota.ts:314]: And in our other calls to retryAsyncWithBackoff

File: packages/phone-number-privacy/combiner/src/database/wrappers/number-pairs.ts [hunk 8]
```diff
@@ -45,7 +45,7 @@ export async function setNumberPairContacts(
     rows.push(data)
   }
   try {
-    await getDatabase().batchInsert(NUMBER_PAIRS_TABLE, rows).timeout(DB_TIMEOUT)
+    await getDatabase().batchInsert(NUMBER_PAIRS_TABLE, rows)
```
Comment (alecps) [inline packages/phone-number-privacy/combiner/src/database/wrappers/number-pairs.ts:48]: We should maybe see if there's another way to time this out

File: packages/phone-number-privacy/signer/src/database/wrappers/account.ts (hunk 6 above)
Comment (alecps) [reply packages/phone-number-privacy/signer/src/database/wrappers/account.ts]: Confirmed this won't cause problems as this code is never used. We should probably remove this code though

File: .env.alfajores (hunk 3 above)
Comment (codyborn) [reply .env.alfajores:296]: doesn't look intentional.. perhaps related to a merge upstream?

File: packages/phone-number-privacy/signer/src/database/wrappers/account.ts [hunk 9]
```diff
@@ -81,27 +98,11 @@ export async function getDidMatchmaking(account: string, logger: Logger): Promis
...
-      newAccount[ACCOUNTS_COLUMNS.didMatchmaking] = new Date()
-      return insertRecord(newAccount)
-    }
-  } catch (err) {
-    Counters.databaseErrors.labels(Labels.update).inc()
-    logger.error(ErrorMessage.DATABASE_UPDATE_FAILURE)
-    logger.error(err)
-    return null
-  }
+export async function getDidMatchmaking(account: string, logger: Logger) {
```
Comment (alecps) [inline packages/phone-number-privacy/signer/src/database/wrappers/account.ts]: Still need to remove this

**Review by codyborn — APPROVED** [review]
LGTM

**Pull request merged**